Source/Core/VideoBackends/D3D12/D3DDescriptorHeapManager.cpp

```diff
@@ -0,0 +1,183 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "VideoBackends/D3D12/D3DBase.h"
+#include "VideoBackends/D3D12/D3DState.h"
+#include "VideoBackends/D3D12/D3DDescriptorHeapManager.h"
+
+namespace DX12
+{
+	bool operator==(const D3DDescriptorHeapManager::SamplerStateSet& lhs, const D3DDescriptorHeapManager::SamplerStateSet& rhs)
+	{
+		// D3D12TODO: Do something more efficient than this.
+		return (!memcmp(&lhs, &rhs, sizeof(D3DDescriptorHeapManager::SamplerStateSet)));
+	}
+
+	D3DDescriptorHeapManager::D3DDescriptorHeapManager(D3D12_DESCRIPTOR_HEAP_DESC *desc, ID3D12Device* device, unsigned int temporarySlots) :
+		ref(1), device(device), currentPermanentOffsetInHeap(0), descriptorHeap(nullptr), descriptorHeapCpuShadow(nullptr)

```

Comment:
This initializer list can omit constant default value assignments here with direct in-body assignments to the class variables.

e.g.

```cpp
class SomeClass
{
private:
    int some_ref = 0;
    int* some_pointer = nullptr;
};
```